[docs] Formatting and proper noun tune-up (#4200)
drud/ddev: docs/content/users/configuration/config_yaml.md

| docroot                          | Relative path to docroot containing `index.php` / `index.html`                            |                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| php_version                      | `5.6`, `7.0`, `7.1`, `7.2`, `7.3`, `7.4`, `8.0`, `8.1`                                    | You can only specify the major version (`7.3`), not a minor version (`7.3.2`), from those explicitly available.                                                                                                                                                                                                                                                                                                                             |

> | webimage                         | Docker image to use for web server                                                        | It’s unusual and not recommended to change the default, but the web image can be overridden with a correctly-crafted image, probably based on [drud/ddev-webserver](https://github.com/drud/ddev/tree/master/containers/ddev-webserver).                                                                                                                                                                                                    |

Author: ```suggestion
| webimage                         | Docker image to use for web server                                                        | It’s unusual and not recommended to change the default, but the web image can be overridden with a correctly-crafted image, probably based on [drud/ddev-webserver](https://github.com/drud/ddev/tree/master/containers/ddev-webserver)—typically, this means additions to the existing web image using a `.ddev/web-build/Dockerfile.*`.                                                                                                                                                                                                    |
```

> | database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |

Author: Wait, can you actually use a single string like `mariadb:5.5` instead of explicit, separate `type: mariadb` + `version: 5.5`? I specifically changed that format because I thought `type` and `version` were required keys and `mariadb:5.5` suggested a format you can’t actually use.

> | database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |

Reviewer: Oh shoot, this is the config_yaml, not the args to ddev config. You're right. But we're going to have to do it better. The "type" key is lower-case, `mariadb` not `MariaDB`.  I see you've got some work done on using HTML to format this right, maybe we can give more examples.

> | database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |

Author: I generally tried to cut characters from this table and make it more consistent since its primary aim is to be a comprehensive reference. The page it links to in this case immediately does a perfect job of illustrating how to specify a database engine and version—no need to try and do that here as well since explanation gets in the way.

It’d be nice to find a format that can neatly accommodate DDEV configuration options _period_, regardless of whether they’re set via YAML or command options—that’s just beyond the scope of immediate cleanup.

> | database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |

Reviewer: This stuff should be able to be generated by the build, instead of being maintained this way.

> | database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |

Author: That’s on my list too! Not necessarily for this page, but an exhaustive list somewhere of the latest DDEV version’s included software and individual versions. Automatically updated for everyone’s joy.

| additional_hostnames             | Array of extra hostnames                                                                  | Example: `additional_hostnames: ["somename", "someothername", "*.thirdname"]` would provide HTTP and HTTPS URLs for `somename.ddev.site`, `someothername.ddev.site`, and `one.thirdname.ddev.site` + `two.thirdname.ddev.site`. The wildcard (`*`) setting only works if you’re using DNS to resolve hostnames (default) and connected to the internet.                                                                                     |
| additional_fqdns                 | Extra fully-qualified domain names                                                        | Example: `additional_fqdns: ["example.com", "sub1.example.com"]` would provide HTTP and HTTPS URLs for `example.com` and `sub1.example.com`. Take care with this because it can cause great confusion and adds extraneous items to your `/etc/hosts` file.                                                                                                                                                                                  |

> | upload_dir                       | Path from docroot to the upload directory targeted by `ddev import-files`                 |                                                                                                                                                                                                                                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| upload_dir                       | Path from docroot to the user-generated-files directory targeted by `ddev import-files`                 |                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```

| omit_containers                  | Containers the project should not load                                                    | Can include `db`, `dba`, and `ddev-ssh-agent`.<br><br>Example: `omit_containers: [db, dba, ddev-ssh-agent]` starts the project without a `db` container, phpMyAdmin, and SSH agent. Some containers can be omitted globally in `~/.ddev/global_config.yaml` and the result is additive; all containers named in both places will be omitted.<br><br>Omitting the `db` container will cause database-dependent DDEV features to be unstable. |
| web_environment                  | Additional custom environment variables for the project’s web container                   | Example: `web_environment: ["SOMEENV=someval", "SOMEOTHERENV=someotherval"]`.                                                                                                                                                                                                                                                                                                                                                               |

> | nfs_mount_enabled                | Whether to use NFS to mount the project into the container for performance                | Can be `true` or `false`, and requires host configuration before it can be used. See [nfs_mount_enabled documentation](../install/performance.md). Project-level configuration of this option is unusual, and when `true` in the global config it overrides the project-specific value.                                                                                                                                                     |

Reviewer: ```suggestion
| nfs_mount_enabled                | Whether to use NFS to mount the project into the container for performance                | Can be `true` or `false`, and requires host configuration before it can be used. See [nfs_mount_enabled documentation](../install/performance.md). Project-level configuration of this option is unusual, and when `true` in the global config it overrides the project-specific value. Note that this is not commonly used now that mutagen is available because mutagen is so much faster.                                                                                                                                                    |
```

> | nfs_mount_enabled                | Whether to use NFS to mount the project into the container for performance                | Can be `true` or `false`, and requires host configuration before it can be used. See [nfs_mount_enabled documentation](../install/performance.md). Project-level configuration of this option is unusual, and when `true` in the global config it overrides the project-specific value.                                                                                                                                                     |

Author: ```suggestion
| nfs_mount_enabled                | Whether to use NFS to mount the project into the container for performance                | Can be `true` or `false`, and requires host configuration before it can be used. See [nfs_mount_enabled documentation](../install/performance.md). Project-level configuration of this option is unusual, and when `true` in the global config it overrides the project-specific value. (Not commonly used now that much-faster Mutagen is available.)                                                                                                                                                    |
```

| dbimage_extra_<wbr>packages      | Extra Debian packages for the project’s database container                                | Example: `dbimage_extra_packages: ["less"]` will add the `less` package when the database container is built.                                                                                                                                                                                                                                                                                                                               |
| use_dns_when_<wbr>possible       | Whether to use DNS instead of editing `/etc/hosts`                                        | Defaults to `true`. When `false`, DDEV will always update the `/etc/hosts` file with the project hostname instead of using DNS for name resolution.                                                                                                                                                                                                                                                                                         |

> | project_tld                      | Default TLD used for project domains                                                      | Defaults to `ddev.site`. Can be changed to anything that works for you. Use `ddev.local` to keep things the way they were before DDEV v1.9.                                                                                                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| project_tld                      | Default TLD used for project domains                                                      | Defaults to `ddev.site`. Can be changed to anything that works for you.                                                                                                                                                                                                                                                                                                 |
```

## global_config.yaml Options

The `$HOME/.ddev/global_config.yaml` file has a few key global config options.

| Option                          | Description                                                                   | Notes                                                                                                                                                                                                                                                                                                                                                                   |
| ------------------------------- | ----------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | nfs_mount_enabled               | Whether to globally enable NFS mounting for all projects                      | Only `true` has any effect. When `true`, NFS will be used on all projects, regardless of any settings in the individual projects.                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| nfs_mount_enabled               | Whether to globally enable NFS mounting for all projects                      | Only `true` has any effect. When `true`, NFS will be used on all projects, regardless of any settings in the individual projects. This is rarely used because mutagen has such widespread usage and is so much faster.                                                                                                                                                                                                                                      |
```

> | nfs_mount_enabled               | Whether to globally enable NFS mounting for all projects                      | Only `true` has any effect. When `true`, NFS will be used on all projects, regardless of any settings in the individual projects.                                                                                                                                                                                                                                       |

Author: ```suggestion
| nfs_mount_enabled               | Whether to globally enable NFS mounting for all projects                      | Only `true` has any effect. When `true`, NFS will be used on all projects, regardless of any settings in the individual projects. (Rarely used since Mutagen is widely used and much faster.)                                                                                                                                                                                                                                       |
```

| internet_detection_timeout_ms   | Internet detection timeout in milliseconds                                    | Defaults to `1000` (1 second). DDEV must detect whether the internet is working to determine whether to add hostnames to `/etc/hosts`. In rare cases you may need to increase this value if you have slow but working internet. See [FAQ](../basics/faq.md) and [GitHub issue](https://github.com/drud/ddev/issues/2409#issuecomment-662448025).                        |
| use_hardened_images             | Whether to use hardened images for internet deployment                        | Defaults to `false`. When `true`, more secure hardened images are used for an internet deployment. These do not include sudo in the web container, and the container is run without elevated privileges. Generally used with the experimental casual hosting feature.                                                                                                   |

> | use_letsencrypt                 | Whether to enable experimental Let’s Encrypt integration                      | Defaults to `false` and may also be set via `ddev global --use-letsencrypt` or `ddev global --use-letsencrypt=false`. When `true`, `letsencrypt_email` must also be set and the system must be available on the internet. Used with experimental casual hosting feature.                                                                                                |

Reviewer: ```suggestion
| use_letsencrypt                 | Whether to enable Let’s Encrypt integration                      | Defaults to `false` and may also be set via `ddev global --use-letsencrypt` or `ddev global --use-letsencrypt=false`. When `true`, `letsencrypt_email` must also be set and the system must be available on the internet. Used with casual webhosting feature.                                                                                                |
```

> | use_letsencrypt                 | Whether to enable experimental Let’s Encrypt integration                      | Defaults to `false` and may also be set via `ddev global --use-letsencrypt` or `ddev global --use-letsencrypt=false`. When `true`, `letsencrypt_email` must also be set and the system must be available on the internet. Used with experimental casual hosting feature.                                                                                                |

Author: Two words:

```suggestion
| use_letsencrypt                 | Whether to enable Let’s Encrypt integration                      | Defaults to `false` and may also be set via `ddev global --use-letsencrypt` or `ddev global --use-letsencrypt=false`. When `true`, `letsencrypt_email` must also be set and the system must be available on the internet. Used with casual web hosting feature.                                                                                                |
```

> | letsencrypt_email               | Email associated with experimental Let’s Encrypt feature                      | Set with `ddev global --letsencrypt-email=me@example.com`. Used with experimental casual hosting feature.                                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| letsencrypt_email               | Email associated with Let’s Encrypt feature                      | Set with `ddev global --letsencrypt-email=me@example.com`. Used with experimental casual hosting feature.                                                                                                                                                                                                                                                               |
```
